app/controllers/users_controller.rb

```diff
@@ -4,7 +4,12 @@ def index
     @users = User.all
   end
 
-  def show; end
+  def show
+    @my_reservations = current_user.borrowed_books
+    @books_reserved_from_me = current_user.books
+                                          .where(status: 'reserved')
```

Comment:
I think `where` accepts an array of values, so you could do `where(status: ['reserved', 'borrowed'])`. Also, it might be good to have a scope in the model for these clauses, maybe `in_use` or `taken`?

Reply:
Also, I think it'd be good to add an enum to the Book` model which defines the statuses that a book can have:
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html

`enum status: [ :available, :reserved, :borrowed]`

Then you can always use the symbol representation of the status to retrieve the relevant records: w.g. `Book.where(status: :reserved)`

Enums are pretty useful in scenarios like this one, where there is a limited set of values that an attribute can have. For example, adding the enum `[ :available, :reserved, :borrowed]` automatically gives you the ability to quickly check if a book is reserved, by doing `book.reserved?`.

Reply:
> Declare an enum attribute where the values map to integers in the database, but can be queried by name.

since I have strings in DB, not integers, I present it as a Hash and it works: 
`enum status: { available: 'available', reserved: 'reserved', borrowed: 'borrowed' }`

Regarding first remark: `enum` provides scopes as well, but for some reason I can't chain them, so smth like: `current_user.books.reserved.borrowed` doesn't work. I think at some point I will want to render `reserved` and `borrowed` books in separate tables and will make use of these scopes, but for now I would leave it like this:
`@books_reserved_or_taken_from_me = current_user.books.where(status: [:reserved, :borrowed])`